### ISSUE TITLE
Implement builder pattern for SpinalConfig

### ISSUE THREAD

**Issue by numero-744** [SpinalHDL/SpinalHDL#881]
I have read several times (in PRs and issues) that `SpinalConfig` would be better with an [builder pattern](https://www.greyblake.com/blog/builder-pattern-in-rust/)-like approach just like `SimConfig`.

**Comment by Dolu1990** [comment]
Yes right, it can feel quite more natural in many cases. Some API can be added
